apps/resource_manager/lib/identity/schemas/client_application.ex

```diff
@@ -83,11 +83,22 @@ defmodule ResourceManager.Identity.Schemas.ClientApplication do
     |> validate_length(:name, min: 1, max: 150)
     |> validate_inclusion(:status, @possible_statuses)
     |> validate_inclusion(:protocol, @possible_protocols)
-    |> validate_inclusion(:grant_flows, @possible_grant_flows)
     |> validate_inclusion(:access_type, @possible_access_types)
     |> unique_constraint(:name)
+    |> validate_grant_flows()
   end
 
+  defp validate_grant_flows(%{valid?: true, changes: %{grant_flows: flows}} = changeset) do
+    if Enum.all?(flows, &(&1 in @possible_grant_flows)) do
+      changeset
+    else
+      opts = [validation: :subset, enum: @possible_grant_flows]
+      add_error(changeset, :grant_flows, "is invalid", opts)
+    end
+  end
```

Comment:
this function is just to give a customized response?

Reply:
Yeah !
It's better to add the error as a changeset validation failure because that's the expected response.
This helps out a lot when we have a default controller fallback that know how to handle changeset errors.